slomonitor/src/main/scala/com.gu.notifications.slos/SloMonitor.scala

```diff
@@ -47,41 +92,76 @@ object SloMonitor {
       s"partition_date = '${sentTime.toLocalDate}'"
 
     s"""
-         |SELECT COUNT(*)
-         |FROM notification_received_${stage.toLowerCase()}
-         |WHERE notificationid = '${notificationId}'
-         |AND $partitionDate
-         |AND DATE_DIFF('second', from_iso8601_timestamp('${sentTime}'), received_timestamp) < 120
-      """.stripMargin
+       |SELECT
+       |  platform,
+       |  SUM(CASE
+       |    WHEN DATE_DIFF('second', from_iso8601_timestamp('$sentTime'), received_timestamp) < 30 THEN 1
+       |    ELSE 0
+       |  END) AS less_than_30,
+       |  SUM(CASE
+       |    WHEN DATE_DIFF('second', from_iso8601_timestamp('$sentTime'), received_timestamp) < 60 THEN 1
+       |    ELSE 0
+       |  END) AS less_than_60,
+       |  SUM(CASE
+       |    WHEN DATE_DIFF('second', from_iso8601_timestamp('$sentTime'), received_timestamp) < 90 THEN 1
+       |    ELSE 0
+       |  END) AS less_than_90,
+       |  SUM(CASE
+       |    WHEN DATE_DIFF('second', from_iso8601_timestamp('$sentTime'), received_timestamp) < 120 THEN 1
+       |    ELSE 0
+       |  END) AS less_than_120,
+       |  SUM(CASE
+       |    WHEN DATE_DIFF('second', from_iso8601_timestamp('$sentTime'), received_timestamp) < 150 THEN 1
+       |    ELSE 0
+       |  END) AS less_than_150,
+       |  SUM(CASE
+       |    WHEN DATE_DIFF('second', from_iso8601_timestamp('$sentTime'), received_timestamp) < 180 THEN 1
+       |    ELSE 0
+       |  END) AS less_than_180,
+       |  SUM(CASE
+       |    WHEN DATE_DIFF('second', from_iso8601_timestamp('$sentTime'), received_timestamp) < 210 THEN 1
+       |    ELSE 0
+       |  END) AS less_than_210,
+       |  SUM(CASE
+       |    WHEN DATE_DIFF('second', from_iso8601_timestamp('$sentTime'), received_timestamp) < 240 THEN 1
+       |    ELSE 0
+       |  END) AS less_than_240,
+       |  SUM(CASE
+       |    WHEN DATE_DIFF('second', from_iso8601_timestamp('$sentTime'), received_timestamp) < 270 THEN 1
+       |    ELSE 0
+       |  END) AS less_than_270,
+       |  SUM(CASE
+       |    WHEN DATE_DIFF('second', from_iso8601_timestamp('$sentTime'), received_timestamp) < 300 THEN 1
+       |    ELSE 0
+       |  END) AS less_than_300,
+       |  COUNT(*) AS total_deliveries
```

Comment:
N.B. that `total_deliveries` really means `total_deliveries_when_this_query_was_run` - in practice that means the number of deliveries [in the first 10 minutes](https://github.com/guardian/mobile-n10n/blob/63dd9b0ca1b3f1020cc65c372c0c72fd1cb31403/notification/app/notification/services/SloTrackingSender.scala#L18), at present.

Reply:
I think this is a good enough for now. I know that we've been keen to explore the number of missing/lost deliveries too, so at some point in the future we may want to update how we define the "total".